Content.Shared/GameObjects/Components/Mobs/State/SharedMobStateComponent.cs

```diff
@@ -0,0 +1,334 @@
+
+        [ViewVariables]
+        public int? CurrentThreshold { get; private set; }
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+
+            serializer.DataReadWriteFunction(
+                "thresholds",
+                new Dictionary<int, IMobState>(),
+                thresholds =>
+                {
+                    _lowestToHighestThresholds = new SortedDictionary<int, IMobState>(thresholds);
+                },
+                () => new Dictionary<int, IMobState>(_lowestToHighestThresholds));
+        }
+
+        protected override void Startup()
+        {
+            base.Startup();
+
+            if (CurrentState != null && CurrentThreshold != null)
+            {
+                UpdateState(null, (CurrentState, CurrentThreshold.Value));
+            }
+        }
+
+        public override void OnRemove()
+        {
+            if (Owner.TryGetComponent(out SharedAlertsComponent? status))
+            {
+                status.ClearAlert(AlertType.HumanHealth);
+            }
+
+            base.OnRemove();
+        }
+
+        public override ComponentState GetComponentState()
+        {
+            return new MobStateComponentState(CurrentThreshold);
+        }
+
+        public override void HandleComponentState(ComponentState? curState, ComponentState? nextState)
+        {
+            base.HandleComponentState(curState, nextState);
+
+            if (!(curState is MobStateComponentState state))
+            {
+                return;
+            }
+
+            if (CurrentThreshold == state.CurrentThreshold)
+            {
+                return;
+            }
+
+            if (state.CurrentThreshold == null)
+            {
+                RemoveState(true);
+            }
+            else
+            {
+                UpdateState(state.CurrentThreshold.Value, true);
+            }
+        }
+
+        public override void HandleMessage(ComponentMessage message, IComponent? component)
+        {
+            base.HandleMessage(message, component);
+
+            switch (message)
+            {
+                case DamageChangedMessage msg:
+                    if (msg.Damageable.Owner != Owner)
+                    {
+                        break;
+                    }
+
+                    UpdateState(msg.Damageable.TotalDamage);
+
+                    break;
+            }
+        }
+
+        public bool IsAlive()
+        {
+            return CurrentState?.IsAlive() ?? false;
+        }
+
+        public bool IsCritical()
+        {
+            return CurrentState?.IsCritical() ?? false;
+        }
+
+        public bool IsDead()
+        {
+            return CurrentState?.IsDead() ?? false;
+        }
+
+        public bool IsIncapacitated()
+        {
+            return CurrentState?.IsIncapacitated() ?? false;
+        }
+
+        public (IMobState state, int threshold)? GetHighestThreshold(int damage)
```

Comment:
In my personal opinion the naming scheme of "thresholds" is weird in general, i.e. for this function I'd prefer just to call it something like "GetCurrentMobState"

Reply:
Changed.